src/fsharp/ConstraintSolver.fs

```diff
@@ -1417,10 +1420,10 @@ and SolveRelevantMemberConstraints (csenv:ConstraintSolverEnv) ndeep permitWeakR
             tps |> AtLeastOneD (fun tp -> 
                 /// Normalize the typar 
                 let ty = mkTyparTy tp
-                if isAnyParTy csenv.g ty then 
-                    let tp = destAnyParTy csenv.g ty
+                match tryAnyParTy csenv.g ty with
```

Comment:
you have done similar changes in separate PR, is it a pattern that keeps happening in code base: one function call for branch, followed by another call, always the two same functions?